### PR TITLE
Fix TypeReferences emitted by OldRod using incorrect CorLib.

### PR DESCRIPTION
**Pull request by ElektroKill**
The previous method would create references to `netstndard` instead of `mscorlib` for .NET Framework executables. 

This does not prevent this problem from occurring when injecting the VmHelper type since the importing there is done by AsmResolver and it doesn't offer any way to easily change these references to the correct core library when injecting. In theory, OldRod could iterate over the injected members to patch the references but that seems rather hacky.

### CODE REVIEW

**Comment by ElektroKill** [comment]
Yes, for .NET Framework 4.6, netstandard references should be fine but for lower frameworks like the ancient 2.0 or 3.5 it may cause issues which is why I made this partial fix.

**Comment by Washi1337** [comment]
Not sure if this is a viable "partial" solution then. Best to tackle this problem at its core rather than trying to bandage some unwanted side-effects that still do not really remove references to netstandard.

File: src/OldRod.Core/Recompiler/Transform/TypeHelper.cs [hunk 1]
```diff
@@ -34,13 +34,13 @@ public class TypeHelper
...
         public TypeHelper(ReferenceImporter importer)
         {
-            var scope = importer.TargetModule.CorLibTypeFactory.CorLibScope;
+            var ownerModule = importer.TargetModule;
+            var factory = ownerModule.CorLibTypeFactory;
+            var scope = ownerModule.CorLibTypeFactory.CorLibScope;
 
-            _arrayType = importer.ImportType(new TypeReference(scope, "System", "Array"));
-            _objectType = importer.ImportType(new TypeReference(scope, "System", "Object"));
+            _arrayType = importer.ImportType(new TypeReference(ownerModule, scope, "System", "Array"));
```
Comment (Washi1337) [inline src/OldRod.Core/Recompiler/Transform/TypeHelper.cs:41]: Why is this changed?

File: src/OldRod.Core/CodeGen/VmHelperGenerator.cs [hunk 2]
```diff
@@ -59,5 +59,20 @@ public static TypeDefinition ImportFlagHelper(ModuleDefinition module, VMConstan
...
+        {
+            internal UseExistingCorlibReferenceImporter(MemberCloneContext context) : base(context) 
+            {
+            }
+
+            protected override ITypeDefOrRef ImportType(TypeReference type)
+            {
+                var defAsm = type.Scope?.GetAssembly();
+                if (defAsm is not null && defAsm.IsCorLib)
+                    type = new TypeReference(Context.Module, TargetModule.CorLibTypeFactory.CorLibScope, type.Namespace, type.Name);
```
Comment (Washi1337) [inline src/OldRod.Core/CodeGen/VmHelperGenerator.cs:73]: This is already an imported type, there's no need to import it again.

File: src/OldRod.Core/Recompiler/VCall/ThrowRecompiler.cs [hunk 3]
```diff
@@ -16,7 +17,9 @@ public CilExpression Translate(RecompilerContext context, ILVCallExpression expr
                 return new CilInstructionExpression(CilOpCodes.Rethrow);
 
             var argument = (CilExpression) expression.Arguments[2].AcceptVisitor(context.Recompiler);
-            argument.ExpectedType = context.ReferenceImporter.ImportType(typeof(Exception));
+
+            var scope = context.TargetModule.CorLibTypeFactory.CorLibScope;
+            argument.ExpectedType = context.ReferenceImporter.ImportType(new TypeReference(context.TargetModule, scope, "System", "Exception"));
```
Comment (Washi1337) [inline src/OldRod.Core/Recompiler/VCall/ThrowRecompiler.cs:22]: This is already an imported type. No need to import it again.

File: src/OldRod.Core/Recompiler/Transform/TypeHelper.cs (hunk 1 above)
Comment (ElektroKill) [reply src/OldRod.Core/Recompiler/Transform/TypeHelper.cs:41]: I changed the code to use the constructor which specifies the owner module so that the `Module` property is set. If I remember correctly this was null previously and it is being used on the inside of the TypeReference class and it was causing an issue on a file I tested on. Since I made this change I have a new PC and I don't have the file anymore :/ sorry.

Also, I think the `ImportType` call may be redundant here too as the TypeReference already has a target module, scope, name, and namespace specified.

Even if we aren't going by the bug fix I mentioned above changing the code to set `Module` property and skipping importing would be faster as the importer would not have to process a scope that is already imported and ready to go.

Comment (Washi1337) [reply src/OldRod.Core/Recompiler/Transform/TypeHelper.cs:41]: My point is (and it applies to all comments), that you should either set the module yourself using the constructor, OR use `ImportType`, not both. Importing a type that is specifically built as an imported reference is redundant.

**Review by Washi1337 — APPROVED** [review]

**Comment by Washi1337** [comment]
Sorry for the very late response. Thanks for adding it to oldrod!
